x-pack/plugins/infra/public/pages/metrics/hosts/components/hosts_table.tsx

```diff
@@ -22,11 +22,11 @@ export const HostsTable = () => {
   const { onSubmit, searchCriteria } = useUnifiedSearchContext();
   const [properties, setProperties] = useTableProperties();
 
-  const { columns, items, isFlyoutOpen, closeFlyout, clickedItemUuid } = useHostsTable(hostNodes, {
+  const { columns, items, isFlyoutOpen, closeFlyout, clickedItemId } = useHostsTable(hostNodes, {
     time: searchCriteria.dateRange,
   });
 
-  const clickedItem = items.find(({ uuid }) => uuid === clickedItemUuid);
+  const clickedItem = items.find(({ id }) => id === clickedItemId);
```

Comment:
**nit**: just realized now that also the current item (e.g. `selectedHost`) is derived from internals of the `useHostTable` hook, you can so lift this computation into the hook and memoize the value of the selected host (say we have 100 hosts, we don't want to iterate all of them on each render if the selection doesn't change), so that the hook directly return the selected host or undefined if not selected

Reply:
Good idea, I moved that to the hook and memoized it 👍